contracts/ClearingHouse.sol

```diff
@@ -848,30 +851,25 @@ contract ClearingHouse is
             0
         );
 
-        if (response.pnlToBeRealized != 0) {
-            // if realized pnl is not zero, that means trader is reducing or closing position
-            // trader cannot reduce/close position if the remaining account value is less than
-            // accountValue * LiquidationPenaltyRatio, which
-            // enforces traders to keep LiquidationPenaltyRatio of accountValue to
-            // shore the remaining positions and make sure traders having enough money to pay liquidation penalty.
+        int256 takerPositionSizeAfterSwap =
+            IAccountBalance(_accountBalance).getTakerPositionSize(params.trader, params.baseToken);
 
-            // CH_NEMRM : not enough minimum required margin after reducing/closing position
-            require(
-                getAccountValue(params.trader) >=
-                    _getTotalAbsPositionValue(params.trader).mulRatio(_getLiquidationPenaltyRatio()).toInt256(),
-                "CH_NEMRM"
-            );
-        }
+        bool hasBecameInversePosition = takerPositionSizeBeforeSwap ^ takerPositionSizeAfterSwap < 0;
+
+        if (response.pnlToBeRealized != 0 && !hasBecameInversePosition) {
```

Comment:
There's a very edge case that the user is reducing position but the `pnlToBeRealized` because of `openNotional * reducingRatio = closedNotional`.

However, I think it won't be an issue because we require a higher margin ratio. How do you think?

Reply:
Ah, you are right, but I think we can't skip it. This will introduce another UX issue and not follow the actual spec.

Let's say when a trader's margin ratio is between < imRatio and > mmRatio. If this trader luckily to just reduce a position with 0 realized pnl (due to funding payment or inital fee get covered), this trader will get reverted.  

I'll switch this if...else... to fulfill this goal.